crates/signature-mpc/src/twopc_mpc_protocols/mod.rs

```diff
@@ -705,3 +705,48 @@ pub fn recovery_id(
         ),
     }
 }
+
+pub fn verify_signatures(
+    messages: Vec<Vec<u8>>,
+    hash: &Hash,
+    public_key: PublicKeyValue,
+    signatures: Vec<Vec<u8>>,
+) -> bool {
+    for (message, signature) in messages.iter().zip(signatures.iter()) {
+        match verify_single_signature(message, signature, public_key, hash) {
+            Ok(_) => {}
+            Err(Error::MaliciousDesignatedDecryptingParty) => return false,
+            // TODO (#135): Understand how to handle different errors that may be returned from [`SignatureThresholdDecryptionParty::verify_decrypted_signature`].
+            _ => return false,
+        }
+    }
+    true
+}
+
+pub fn verify_single_signature(
+    message: &Vec<u8>,
+    signature: &Vec<u8>,
+    public_key: PublicKeyValue,
+    hash: &Hash,
+) -> Result<()> {
+    let message = message_digest(message.as_slice(), hash);
+    let (r, s) =
+        bcs::from_bytes::<(Scalar, Scalar)>(signature).map_err(|_| Error::InvalidParameters)?;
+    let public_key = Secp256K1GroupElement::new(
+        public_key,
+        &group::PublicParameters::<Secp256K1GroupElement>::default(),
+    )?;
+    SignatureThresholdDecryptionParty::verify_decrypted_signature(r, s, message, public_key)?;
+    Ok(())
+}
+
+pub fn convert_signature_to_canonical_form(signature: Vec<u8>) -> std::result::Result<Vec<u8>, ()> {
```

Comment:
1. document the func.
2. map the error inside the func and make Result use the error. see: https://rust-lang.github.io/rust-clippy/master/index.html#/result_unit_err